test/framework/include/test_framework/models/Tx.h

```diff
@@ -70,10 +70,16 @@ class Tx
         : m_pTransaction(pTransaction) { }
 
     static Tx CreatePegIn(const uint64_t amount)
+    {
+        BlindingFactor outputBF;
+        return CreatePegIn2(amount, outputBF);
+    }
+
+    static Tx CreatePegIn2(const uint64_t amount, BlindingFactor& outputBF)
```

Comment:
I realize that bitcoin/litecoin use out parameters religiously, but I've avoided them throughout libmw since they're not self-documenting. In the rare cases where I need to return multiple values, I've preferred tuples or custom structs over out parameters.

Let's use this function as an example why. Assume you've never seen this code before, and you're trying to consume `Tx CreatePegIn2(const uint64_t amount, BlindingFactor& outputBF)`. Does it matter what value you pass in as outputBF? Is it purely an output parameter, or is the function's behavior affected by the value of outputBF when it's passed in? Without looking at the code, it's entirely plausible to think that outputBF will only be randomized if it's currently 0. We know from the code that's not the case here, but it *could* be. And without reading the code, it's impossible to tell.

I realize this is just test code, but as we build out our test cases, I expect this class to be used quite heavily. My preferred approach for this would be to just add a vector of `test::TxOutput`s to `test::Tx`, and update `Builder` to populate it.

I'll go ahead and merge this as is, but I wanted to explain my reasoning so you're not offended if you see me make changes to this function later. What you did was perfectly reasonable and common C++ practice, so I didn't want you to think you were doing anything wrong. I'm just weird and picky 😄